pkg/testutils/jobutils/jobs_verification.go

```diff
@@ -82,10 +82,19 @@ func waitForJobToHaveStatus(
 ) {
 	t.Helper()
 	testutils.SucceedsWithin(t, func() error {
+		query := fmt.Sprintf("SELECT status, payload FROM (%s)", InternalSystemJobsBaseQuery)
+		rows, err := db.DB.QueryContext(context.Background(), query, jobID)
+		if err != nil {
+			return err
+		}
 		var status string
 		var payloadBytes []byte
-		query := fmt.Sprintf("SELECT status, payload FROM (%s)", InternalSystemJobsBaseQuery)
-		db.QueryRow(t, query, jobID).Scan(&status, &payloadBytes)
+		if !rows.Next() {
+			return errors.Errorf("no rows returned")
```

Comment:
I suppose we could check rows.Err() first. Not really sure if you'd ever see a non-nil error case on the first row.